include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -485,6 +489,9 @@ struct __memobj_traits<_T*>
 
 } // namespace __internal
 
+template <typename _ExecutionPolicy, typename... _Args>
+using __buffer = __internal::__buffer_impl<::std::decay_t<_ExecutionPolicy>, _Args...>;
```

Comment:
Since it's our API to `__buffer` I believe that having `typename... Args` is incorrect. 

The API should provide as less ways to use it incorrectly is possible. So my suggestion is
```suggestion
template <typename _ExecutionPolicy, typename _T, typename _Container = sycl::buffer<_T, 1>>
using __buffer = __internal::__buffer_impl<::std::decay_t<_ExecutionPolicy>, _T, _Containter>;
```

Having that changed we also need to change the `__buffer_impl` primary template declaration. Please see the comment there.

Reply:
Fixed